packages/es-components/src/components/controls/switch/Switch.js

```diff
@@ -0,0 +1,253 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { noop } from 'lodash';
+
+const Directions = {
+  'top': 'column-reverse',
+  'bottom': 'column',
+  'right': 'row',
+  'left': 'row-reverse'
+}
+
+const Padding = {
+  'top': '0 0 0.5rem',
+  'bottom': '2.25rem 0 0',
+  'right': '0 0 0.5rem',
+  'left': '0 1rem 0 0'
+}
+
+const primary = "#0073b6";
+const primaryLight = "#ededed";
+const success = "#298544";
+const successLight = "#69aa7c";
+const warning = "#c25400";
+const warningLight = "#d4874d";
+const danger = "#cc0000";
+const dangerLight = "#db4d4d";
+
+function Type(type, light) {
+  switch (type) {
+    case "primary":
+      return light ? primaryLight : primary;
+    case "success":
+      return light ? successLight : success;
+    case "warning":
+      return light ? warningLight : warning;
+    case "danger":
+      return light ? dangerLight : danger;
+    default:
+      return light ? primaryLight : primary;
+  }
+}
+
+const SwitchBase = styled.div`
+  margin: 0 auto 10px;
+  display: flex;
+`;
+
+const SwitchLabel = styled.label`
+  flex-direction: ${props => Directions[props.direction]};
+  text-align: ${props => props.direction === 'top' || props.direction === 'bottom' ? 'center' : 'left'};
+  justify-content: ${props => props.direction === 'left' ? 'flex-end' : 'normal'};
+  cursor: pointer;
+  display: flex;
+  position: relative:
+  padding: 5px 0;
+  transition: all .25s linear;
+
+  input[type="checkbox"] {
```

Comment:
It still seems like there are a few things that could be moved into or closer to their styled-components. I *think* this section could be moved to `SwitchInput` without issues.